src/smart-components/role/add-role-new/add-permissions.js

```diff
@@ -94,9 +113,46 @@ const AddPermissionsTable = ({ selectedPermissions, setSelectedPermissions, ...p
       dispatch(fetchRole(baseRoleUuid));
     }
 
+    debounbcedGetApplicationOptions = debounce(
+      ({ applications, resources, operations }) =>
+        fetchOptions({ field: 'application', limit: 50, application: applications.join(), resourceType: resources.join(), verb: operations.join() }),
```

Comment:
You should use `debouncePromise ` from `@redhat-cloud-services/frontend-components-utils` instead of `debounce`